### PR TITLE
Allow unmarshaling ci-operator step graph

### PR DESCRIPTION
**Pull request by alvaroaleman**
Includes https://github.com/openshift/ci-tools/pull/1458 to avoid needing another rebase

The json unmarshaler refuses to unmarshal into a non-empty interface as it can't know the underlying type. So this adds a custom unmarshaler, that unmarshals the manifests into `*unstructured.Unstructured`. This in turn requires TypeMeta which is not set by default, so the second commit adds that.

### CODE REVIEW

File: pkg/api/graph.go [hunk 1]
```diff
@@ -470,22 +472,65 @@ func mergeSteps(into, from CIOperatorStepWithDependencies) CIOperatorStepWithDep
...
 	StartedAt    *time.Time                 `json:"started_at"`
 	FinishedAt   *time.Time                 `json:"finished_at"`
 	Duration     *time.Duration             `json:"duration,omitempty"`
-	Manifests    []ctrlruntimeclient.Object `json:"manifests"`
+	Manifests    []ctrlruntimeclient.Object `json:"manifests,omitempty"`
 	LogURL       string                     `json:"log_url,omitempty"`
 	Failed       *bool                      `json:"failed,omitempty"`
 }
 
+func (c *CIOperatorStepDetailInfo) UnmarshalJSON(data []byte) error {
```
Comment (alvaroaleman) [inline pkg/api/graph.go:499]: If someone can come up with a less inefficient way to do this, that would be great

**Comment by alvaroaleman** [comment]
Ref https://issues.redhat.com/browse/DPTP-1703

**Review by stevekuznetsov — APPROVED** [review]
Confusing and inefficient but we do it once per execution so I think we're OK...

**Pull request merged**